tests/forms_tests/tests/test_forms.py

```diff
@@ -4795,17 +4796,60 @@ def __init__(self, *args, **kwargs):
         with self.assertRaises(KeyError):
             f["name"]
 
+    def test_form_custom_bound_field(self):
+        class TestForm(Form):
+            name = CharField(max_length=10)
+            colour = CharField(max_length=10)
+
+        f = TestForm(bound_field_class=BoundFieldWithoutColon)
```

Comment:
We should also add a test for setting the `bound_field_class` on the class itself. 

``` python
class TestForm(Form):
        name = CharField(max_length=10)
        colour = CharField(max_length=10)
        bound_field_class = BoundFieldWithoutColon
```